jsgrad/models/whisper.ts

```diff
@@ -524,11 +525,45 @@ const wav = (bytes: Uint8Array) => {
   return { sampleRate, numChannels, bitsPerSample, channelData }
 }
 
-export const load_file_waveform = async (filename: string): Promise<Float32Array[]> => {
+function linearInterpolate(data: Float32Array, inputRate: number, outputRate: number): Float32Array {
+  const inputLength = data.length
+  const outputLength = Math.floor(inputLength * (outputRate / inputRate))
+  const outputData = new Float32Array(outputLength)
+  const rateRatio = inputRate / outputRate
+
+  for (let i = 0; i < outputLength; i++) {
+    const inputIndex = i * rateRatio
+    const indexPrev = Math.floor(inputIndex)
+    const indexNext = Math.min(indexPrev + 1, inputLength - 1)
+    const fraction = inputIndex - indexPrev
+
+    const samplePrev = data[indexPrev]
+    const sampleNext = data[indexNext]
+
+    outputData[i] = samplePrev + (sampleNext - samplePrev) * fraction
+  }
+  return outputData
+}
+
+function mergeChannels(channels: Float32Array[]): Float32Array {
+  const length = channels[0].length
+  const merged = new Float32Array(length)
+  for (let i = 0; i < length; i++) {
+    merged[i] = channels.reduce((sum, channel) => sum + channel[i], 0) / channels.length
+  }
+  return merged
+}
+
+export const load_file_waveform = async (filename: string): Promise<Float32Array> => {
   const data = await env.readFile(filename)
   const res = wav(data)
-  if (res.sampleRate !== RATE) throw new Error()
-  return res.channelData
+  const channelData = res.numChannels > 1 ? mergeChannels(res.channelData) : res.channelData[0]
+  if (res.sampleRate !== RATE) {
+    const newSampleRate = RATE
+    const newData = linearInterpolate(channelData, res.sampleRate, newSampleRate)
+    return newData
+  }
```

Comment:
... and here, this could just be 
```suggestion
  if (res.sampleRate !== RATE) return linearInterpolate(channelData, res.sampleRate, RATE)
```